packages/@jsii/kernel/src/kernel.ts

```diff
@@ -114,13 +115,12 @@ export class Kernel {
     }
 
     // read .jsii metadata from the root of the package
-    const jsiiMetadataFile = path.join(packageDir, spec.SPEC_FILE_NAME);
-    if (!fs.pathExistsSync(jsiiMetadataFile)) {
-      throw new Error(
-        `Package tarball ${req.tarball} must have a file named ${spec.SPEC_FILE_NAME} at the root`,
-      );
+    let assmSpec;
+    try {
+      assmSpec = loadAssemblyFromPath(packageDir);
+    } catch (e: any) {
+      throw new Error(`Error for package tarball ${req.tarball}: ${e.message}`);
```

Comment:
Good catch, @RomainMuller, is this better?

Reply:
Yap